include/aspect/material_model/interface.h

```diff
@@ -710,6 +735,15 @@ namespace aspect
          * @}
          */
 
+        /**
+         * If this material model can produce additional named outputs
+         * that are derived from NamedAdditionalOutputs, create them in here.
+         * By default, this does nothing.
+          */
+        virtual
+        void
+        create_additional_named_outputs (MaterialModelOutputs &outputs) const;
```

Comment:
I don't like this function here. It suggests that the material model would create *everything* it knows how to fill. But that may not be necessary. Think of material models that know about the derivatives of coefficients (e.g., for Newton schemes). Should they create the additional outputs? They would then feel compelled to also fill them, even though this may not actually be necessary for what is currently being done with the material model outputs.